Monte_carlo.cpp

```diff
@@ -0,0 +1,324 @@
+/*
+ @file      Monte_carlo.cpp
+ @brief     Utility functions for monte carlo neutron simulation
+ @author    Luke Eure
+ @date      January 12 2016
+*/
+
+#include "Monte_carlo.h"
+
+/*
+ @brief     Generates and transports neutron histories, calculates the mean
+            crow distance
+ @param     n_histories number of neutron histories to run
+ @param     mat a Material object containing information
+            about the material
+ @param     bounds a Boundaries object containing the limits of the
+            bounding box
+ @param     mesh a Mesh object containing information about the mesh
+ @param     num_batches the number of batches to be tested
+*/
+void generateNeutronHistories(int n_histories, Boundaries bounds,
+        Mesh &mesh, int num_batches) {
+
+    // create arrays for tallies and fissions
+    std::vector <Tally> tallies(5);
+    std::vector <std::vector <double> >* old_fission_bank = 
+        new std::vector <std::vector <double> >;
+    std::vector <std::vector <double> >* new_fission_bank = 
+        new std::vector <std::vector <double> >;
+    std::vector <std::vector <double> > temp_fission_bank;
+    
+    bool first_round = true;
+
+    for (int batch=1; batch <= num_batches; ++batch) {
+
+        // clear flux data
+        mesh.fluxClear();
+
+        // assign new fission locations to old fission locations
+        *old_fission_bank = *new_fission_bank;
+        *new_fission_bank = temp_fission_bank;
```

Comment:
This will actually still do a deep copy. Instead, let's do:
```
std::vector <std::vector <double> >* temp_fission_bank;
temp_fission_bank = old_fission_bank;
old_fission_bank = new_fission_bank;
new_fission_bank = temp_fission_bank;
new_fission_bank.clear();
```